controllers/k8ssandra/medusa_reconciler_test.go

```diff
@@ -80,6 +85,30 @@ func createMultiDcClusterWithMedusa(t *testing.T, ctx context.Context, f *framew
 				CassandraUserSecretRef: corev1.LocalObjectReference{
 					Name: cassandraUserSecret,
 				},
+				ReadinessProbe: &corev1.Probe{
```

Comment:
Question: why are we fully specifying so much config here? What is the failure we're trying to catch?

Reply:
From my experience, if you're worried about the way this stuff makes its way from the CR to the output resources, you're best off using an e2e test, since the API server can play tricks on you that won't be caught in an envtest.

Reply:
I think an envtest is OK in this case because you aren't doing anything that really relies on the API server (e.g. defaults, etc.) but given that you aren't relying much on the API server, I think you could simplify the test by not specifying a lot of these settings.